### PR TITLE
feat(java): Allow overrides of package prefixes

### PR DESCRIPTION
**Pull request by ajgateno**
Java SDK generator uses `com.{dot-separated-organization-name}` as the package prefix. This PR adds the ability to override the package prefix from the `java-sdk` config group.

### CODE REVIEW

File: seed/java-sdk/java-custom-package-prefix/java-custom-package-prefix/src/main/java/com/customprefix/resources/imdb/types/Movie.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/**
+ * This file was auto-generated by Fern from our API Definition.
+ */
+package com.customprefix.resources.imdb.types;
```
Comment (amckinney) [inline seed/java-sdk/java-custom-package-prefix/java-custom-package-prefix/src/main/java/com/customprefix/resources/imdb/types/Movie.java:4]: Nice 👏

File: generators/java/sdk/src/main/java/com/fern/java/client/Cli.java [hunk 2]
```diff
@@ -119,8 +119,12 @@ public void runInGithubModeHook(
...
             JavaSdkCustomConfig customConfig,
             GithubOutputMode githubOutputMode) {
-        ClientPoetClassNameFactory clientPoetClassNameFactory = new ClientPoetClassNameFactory(
-                AbstractPoetClassNameFactory.getPackagePrefixWithOrgAndApiName(ir, generatorConfig.getOrganization()));
+        List<String> packagePrefixTokens = customConfig
+                .packagePrefix()
+                .map(List::of)
+                .orElseGet(() -> AbstractPoetClassNameFactory.getPackagePrefixWithOrgAndApiName(
+                        ir, generatorConfig.getOrganization()));
+        ClientPoetClassNameFactory clientPoetClassNameFactory = new ClientPoetClassNameFactory(packagePrefixTokens);
```
Comment (amckinney) [inline generators/java/sdk/src/main/java/com/fern/java/client/Cli.java:127]: Nice and simple 👍

**Review by amckinney — APPROVED** [review]
Nicely done! Let's add an entry to the [versions.yml](https://github.com/fern-api/fern/blob/7d877b6aab90ceab784ee81593583ea7150926ce/generators/java/sdk/versions.yml#L1) so that this gets released. Given that it's a new configuration option, we can release a new minor (i.e. `2.4.0`).

File: packages/cli/ete-tests/src/tests/generate/__snapshots__/generate.test.ts.snap [hunk 3]
```diff
@@ -1,7 +1,3 @@
 // Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
 
-exports[`fern generate > missing docs page 1`] = `
-"[docs]: Found 1 errors and 0 warnings. Run fern check --warnings to print out the warnings.
-[docs]: docs.yml -> navigation -> 0 -> path
-        Path missing.md does not exist"
-`;
```
[the remark this answers is not in the document]
Comment (ajgateno) [reply packages/cli/ete-tests/src/tests/generate/__snapshots__/generate.test.ts.snap]: Just reset this to match main

**Pull request merged**